### PR TITLE
6 get api article id comments

### PR DESCRIPTION
**Pull request by ITedInnovator**

### CODE REVIEW

File: models/comments.models.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 const db = require("../db/connection");
+const { selectArticle } = require("./articles.models");
 
 exports.selectCommentsByArticleId = (article_id) => {
     const sql = `SELECT * FROM comments WHERE article_id = $1 
     ORDER BY created_at DESC;`;
 
-    return db.query(sql, [article_id]).then( ({rows}) => {
+    return selectArticle(article_id).then(({rows}) => {
```
Comment (dappernerddesigns) [inline models/comments.models.js:8]: This is one way to handle this. In other servers you may find it useful to extract out a function that checks if a particular row exists in any table. But if your tests are passing, nicely done.

**Review by dappernerddesigns — APPROVED** [review]
Nice work, I'd like you to add an expect for a length on your comments. When using a forEach we should also assert that we are certain we are getting something back from the database before we check everything.

Good effort and a cheeky reuse of a model function, well done.

File: models/comments.models.js (hunk 1 above)
Comment (ITedInnovator) [reply models/comments.models.js:8]: Good point having a utility function has its advantages. It wouldn't reduce the number of queries though would it? I guess creating a ref object/ array could be something to try later.

Comment (cjplatten) [reply models/comments.models.js:8]: It is the same number of api calls but it is helpful to have a function where confirming an article exisits is its specific purpose to allow for for clarity and control.

**Review by cjplatten — CHANGES_REQUESTED** [review]
Please see comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -299,6 +299,8 @@ describe("Comment endpoints", () => {
                 const { body } = res;
                 expect(Array.isArray(body.comments)).toBe(true);
 
+                expect(body.comments.length).toBe(0);
```
[the remark this answers is not in the document]
Comment (ITedInnovator) [reply __tests__/app.test.js:302]: Hi 6 comments are in the dev data but not the tests. I've changed the endpoint now. Thank you!

**Pull request merged**